### PR TITLE
[Refactor] Python3 upgrade and Docker containerization

### PR DESCRIPTION
**Pull request by rchenzheng**
- add support for python3
- add interactive generation of avahi service files through docker
- convert readme to markdown

No longer will you need to check for dependencies when generating the avahi service file for AirPrint

Thanks @AlexMorro for the issue https://github.com/tjfontaine/airprint-generate/issues/23

Testing methodology

- [x] Generate avahi service file
- [x] Compare with working file in the live server that was created with python2
- [x] Create VM and generate an avahi service file using python2 and compare with python3 generated file
- [x] Test print from iPhone

### CODE REVIEW

**Comment by tjfontaine** [comment]
Ok, a couple thoughts:

It would be nice to split out the changes from #23 with @AlexMorro as the author, and @rchenzheng as the committer — just for clean ownership tracking purposes. Then have the `Dockerfile` addition as its own commit.

Second, I’m quite likely to archive this project after landing this PR, given I don’t maintain any environments that need this functionality — so let me know if you want to maintain your repo and I’ll link to you from the readme.

**Comment by rchenzheng** [comment]
> Ok, a couple thoughts:
> 
> It would be nice to split out the changes from #23 with @AlexMorro as the author, and @rchenzheng as the committer — just for clean ownership tracking purposes. Then have the `Dockerfile` addition as its own commit.
> 
> Second, I’m quite likely to archive this project after landing this PR, given I don’t maintain any environments that need this functionality — so let me know if you want to maintain your repo and I’ll link to you from the readme.

I merely took inspiration from it, but credit where credit is due if @AlexMorro wants to create his own PR. I've tested and written the python3 upgrade in docker

I'd be willing to maintain it but I merely use it on a hobby scale.

**Comment by AlexMorro** [comment]
I just changed the imports to work with python 3 and uploaded it if other people had the same issue. I don't have the time and knowledge for maintaining it, at least for now, I don't wan't the credit or anything.
@rchenzheng you are welcome for the inspiration, I'm glad.

**Comment by rchenzheng** [comment]
@tjfontaine Can you review and merge?

**Review by tjfontaine — APPROVED** [review]
I went back and forth on whether we should defensively import for folks who may be stuck on python2, though I suspect they can either look at the git history, or leave it as an exercise for the person looking to run it.

**Pull request merged**
